Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -104,205 +206,432 @@ if (-not($exchangeShell.ShellLoaded)) {
+                                    }
+                                }
+                            } else {
+                                $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                Write-SimpleLogFile -String ("[FAIL] - Unsigned Module File $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                            }
+                        } else {
+                            if ( $signature.Status -eq 'Valid' -and $signature.StatusMessage -eq 'Signature verified.') {
+                                $cert = $null
+                                $cert = $signature.SignerCertificate
+                                if ( $null -eq $cert) {
+                                    $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                    Write-SimpleLogFile -String ("[FAIL] - We could not get signer certificate on Module: $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                                } else {
+                                    $chain = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Chain
+                                    $chain.Build($cert) | Out-Null
+                                    $rootCertificate = $null
+                                    $rootCertificate = $chain.ChainElements[$chain.ChainElements.Count - 1].Certificate.Subject
+                                    if ( $null -eq $rootCertificate) {
+                                        $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                        Write-SimpleLogFile -String ("[FAIL] - We could not get root certificate on Module: $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                                    } else {
+                                        $FIPSPath = Join-Path $env:ExchangeInstallPath 'FIP-FS\Bin\TE'
```

Comment:
Using `$env:ExchangeInstallPath` but not a guarantee that it is there. We already define the value in `$ExchangePath` we should be using that.